src/inc/configuration-document.ts

```diff
@@ -0,0 +1,116 @@
+import defu from 'defu'
+import type { Scheme, OpenIDConnectConfigurationDocument } from '../index'
+import Storage from '../core/storage'
+
+class ConfigurationDocumentRequestError extends Error {
+  constructor() {
+    super('Error loading OpenIDConnect configuration document')
+    this.name = 'ConfigurationDocumentRequestError'
+  }
+}
+
+// eslint-disable-next-line no-console
+const ConfigurationDocumentWarning = (message: string) =>
+  // eslint-disable-next-line no-console
+  console.warn(`[AUTH] [OPENID CONNECT] Invalid configuration. ${message}`)
+
+export default class ConfigurationDocument {
```

Comment:
Could you write a short comment explaining what configuration doc is, what it does, and why it's necessary?